bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,325 @@
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly-maven-plugin[WildFly maven plugins]
+
+* https://github.com/wildfly/quickstart[WildFly Quickstarts]
+
+* https://start.microprofile.io/[Microprofile starter]
+
+
+== Requirements
+
+==== Bootable jar server vs Standalone server
+
+Although the bootable jar can be seen as a standalone server the following limitations apply:
+
+* CLI management operations that would imply a server restart are not supported.
+* :shutdown operation with restart (eg: by sending CLI operation :shutdown(restart=true)) is not supported. 
+  A warning is displayed and the bootable jar process is terminated.
+* Changes done to the management model (eg: using CLI or Web console) are not persisted between bootable jar executions.
+
+==== Defining a new WildFly Maven Plugin
+
+Existing maven plugins related to provision, build, and manage WildFly:
+
+* https://github.com/wildfly/galleon[org.jboss.galleon:galleon-maven-plugin]
+** Maven plugin goals to provision server are defined in the wildfly/galleon project
+* https://github.com/wildfly/galleon-plugins[org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin]
+** Maven plugin goals to build galleon feature-packs for WildFly are defined in wildfly/galleon-plugins project
+* https://github.com/wildfly/wildfly-maven-plugin[org.wildfly.plugins:wildfly-maven-plugin]
+** Maven plugin goals to manage a WildFly server and its deployments are defined in wildfly/wildfly-maven-plugin project
+
+It makes sense to define a new maven artifact for the plugin and not extend an existing plugin:
+
+* The new plugin depends on galleon-maven-plugin for the provisioning aspects but is WildFly specific. Galleon is only one aspect of it.
+* Adding goals to wildfly-maven-plugin would create confusion (start/run/shutdown are already existing goals that would need to be named differently for the bootable jar). The deployment model is also different.
+
+So the new plugin should be a new Maven artifact with the same groupID as the wildfly-maven-plugin: org.wildfly.plugins, artifactId: wildfly-jar-maven-plugin
+A new GIT project in wildfly organization:  wildfly-jar-maven-plugin
+
+The bootable jar maven project is currently hosted in https://github.com/wildfly-extras/wildfly-jar-maven-plugin[wildfly-extras/wildfly-jar-maven-plugin]
+
+=== Hard Requirements
+
+For a complete list of plugin options, check the plugin https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/download/2.0.0.Alpha4/index.html[community documentation].
+
+==== Build the bootable jar
+
+A new maven plugin and goal (wildfly-jar:package) is developed to output a bootable jar during maven build.
+The bootable jar is composed of:
+
+* Zipped server provisioned thanks to galleon (slim or fat server).
+* User application (war, jar, ear) if not a hollow bootable jar.
```

Comment:
I'd reword this sentence to let the reader understand that the user application is supposed to be included _unless_ a hollow jar is created.
Moreover, I am not sure I understand how this sentence matches what at https://github.com/wildfly/wildfly-proposals/pull/306/files#diff-7278f7017baf3c3aeae30db0b61dec89R148 and in the `Hollow JAR` section.
I mean that when reading this very sentence I understand that a `Hollow JAR` is a `Bootable JAR` _without_ any user application, so basically a server will run.
Instead when I read the contents in the following sections I understand that the user application will be zipped (hence will be in the JAR) but not deployed. 
I think there's need for this aspect to be clarified. WDYT?

Reply:
Clarified that no deployment is present in hollow jar.